BuildResidentialHPXML/measure.rb

```diff
@@ -3772,9 +3772,13 @@ def argument_errors(args)
 module HPXMLFile
   # create the closed-form geometry, and then call individual set_xxx methods
   def self.create(runner, model, args, epw_path, hpxml_path, existing_hpxml_path)
-    epw_file = OpenStudio::EpwFile.new(epw_path)
-    if (args[:hvac_control_heating_season_period].to_s == HPXML::BuildingAmerica) || (args[:hvac_control_cooling_season_period].to_s == HPXML::BuildingAmerica) || (args[:apply_defaults])
-      weather = WeatherProcess.new(epw_path: epw_path, runner: nil)
+    if (args[:hvac_control_heating_season_period].to_s == HPXML::BuildingAmerica) ||
+       (args[:hvac_control_cooling_season_period].to_s == HPXML::BuildingAmerica) ||
+       (args[:solar_thermal_system_type] != 'none' && args[:solar_thermal_collector_tilt].start_with?('latitude')) ||
+       (args[:pv_system_present] && args[:pv_system_array_tilt].start_with?('latitude')) ||
+       (args[:pv_system_2_present] && args[:pv_system_2_array_tilt].start_with?('latitude')) ||
+       (args[:apply_defaults])
+      weather = WeatherFile.new(epw_path: epw_path, runner: nil)
```

Comment:
Put this stuff in its own method that returns true or false, and then just call something like `if need_weather_based_on_args(args) ...`?

Reply:
Yeah, I got a little lazy here...

Really it would be better to obtain the `weather` object on an as-need basis (and only if it hasn't already been obtained). Let me take a quick stab at that.

Reply:
Meh, it's getting more complicated than I like. I'll just go with your suggestion.